openedx/core/djangoapps/user_api/accounts/tests/test_retirement_views.py

```diff
@@ -0,0 +1,1492 @@
+        """
+        statuses = self.create_partner_reporting_statuses()
+        self.assert_status_and_count(statuses, remaining_count=0)
+
+    def test_no_usernames(self):
+        """
+        Checks that if no usernames are passed in we will get a 400 back.
+        """
+        statuses = self.create_partner_reporting_statuses()
+        self.assert_status_and_count([], len(statuses), expected_status=status.HTTP_400_BAD_REQUEST)
+
+    def test_username_does_not_exist(self):
+        """
+        Checks that if a username is passed in that does not have a UserRetirementPartnerReportingStatus row
+        we will get a 400 back.
+        """
+        statuses = self.create_partner_reporting_statuses()
+        orig_count = len(statuses)
+
+        # Create a bogus user that has a non-saved row. This user doesn't exist in the database, so
+        # it should trigger the "incorrect number of rows" error.
+        user = UserFactory()
+        statuses.append(
+            UserRetirementPartnerReportingStatus(
+                user=user,
+                original_username=user.username,
+                original_email=user.email,
+                original_name=user.first_name + ' ' + user.last_name,
+                is_being_processed=True
+            )
+        )
+        self.assert_status_and_count(statuses, orig_count, expected_status=status.HTTP_400_BAD_REQUEST)
+
+    def test_username_in_wrong_status(self):
+        """
+        Checks that if an username passed in has the wrong "is_being_processed" value we will get a 400 error.
+        """
+        # Create some status rows in the expected status
+        statuses = self.create_partner_reporting_statuses()
+
+        # Create some users in the wrong processing status, should trigger the "incorrect number of rows" error.
+        statuses += self.create_partner_reporting_statuses(is_being_processed=False)
+
+        self.assert_status_and_count(statuses, len(statuses), expected_status=status.HTTP_400_BAD_REQUEST)
+
+    def test_does_not_delete_users_in_process(self):
+        """
+        Checks that with mixed "is_being_processed" values in the table only the usernames passed in will
+        be deleted.
+        """
+        statuses = self.create_partner_reporting_statuses()
+
+        self.create_partner_reporting_statuses(is_being_processed=False)
+        self.assert_status_and_count(statuses, len(statuses))
+
+
+@unittest.skipUnless(settings.ROOT_URLCONF == 'lms.urls', 'Account APIs are only supported in LMS')
+class TestPartnerReportingList(ModuleStoreTestCase):
+    """
+    Tests the partner reporting list endpoint
+    """
+
+    def setUp(self):
+        super(TestPartnerReportingList, self).setUp()
+        self.test_superuser = SuperuserFactory()
+        self.course = CourseFactory()
+        self.course_awesome_org = CourseFactory(org='awesome_org')
+        self.courses = (self.course, self.course_awesome_org)
```

Comment:
Is testing with a single org enough?

Reply:
Nope, that's why I test with 2. :) We get the default one from `self.course`, and the second course has one explicitly created to be different.

Reply:
:oops: